File: src/main/java/io/pivotal/example/restcallscheduled/RestCallScheduledApplication.java
package io.pivotal.example.restcallscheduled;

import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.context.annotation.Bean;
import org.springframework.scheduling.annotation.EnableScheduling;
import org.springframework.web.client.RestTemplate;

@SpringBootApplication
@EnableScheduling
class RestCallScheduledApplication {

    public static void main(String[] args) {
        SpringApplication.run(RestCallScheduledApplication.class);
    }

    @Bean
    public RestTemplate restTemplate() {
        return new RestTemplate();
    }

}

File: src/main/kotlin/io/pivotal/example/restcallscheduled/RestCallScheduledApplication.kt
package io.pivotal.example.restcallscheduled

import org.springframework.beans.factory.annotation.Value
import org.springframework.boot.SpringApplication
import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.context.annotation.Bean
import org.springframework.scheduling.annotation.EnableScheduling
import org.springframework.web.client.RestTemplate

@SpringBootApplication
@EnableScheduling
class RestCallScheduledApplication {

    @Bean
    fun restTemplate() = RestTemplate()

}

fun main(args: Array<String>) {
    SpringApplication.run(RestCallScheduledApplication::class.java, *args)
}
